File: app/src/main/java/com/info/zhangxiaolong/myapp/main/nav/NavMenu.java
package com.info.zhangxiaolong.myapp.main.nav;


import java.util.ArrayList;
import java.util.List;

/**
 * Created by zhangxiaolong on 2018/2/20.
 */

public class NavMenu {
    private final List<NavMenuItem> items = new ArrayList<>();

    public void addMenuItem(NavMenuItem item) {
        if(item != null) {
            items.add(item);
        }
    }

    public void clear() {
        items.clear();
    }

    public List<NavMenuItem> getItems() {
        return new ArrayList<>(items);
    }
}

File: app/src/main/java/com/info/zhangxiaolong/myapp/main/nav/NavMenu.kt
package com.info.zhangxiaolong.myapp.main.nav


import java.util.ArrayList

/**
 * Created by zhangxiaolong on 2018/2/20.
 */

class NavMenu {
    private val items = ArrayList<NavMenuItem>()

    fun addMenuItem(item: NavMenuItem?) {
        if (item != null) {
            items.add(item)
        }
    }

    fun clear() {
        items.clear()
    }

    fun getItems(): List<NavMenuItem> {
        return ArrayList(items)
    }
}
